### PR TITLE
Handle NaN correctly in bindings

### LINKED ISSUE #1799: Data binding causes infinite loop if value is NaN
Steps to reproduce:

1. Create two elements and bind some property.
2. Make value NaN
4. Notice error: `Uncaught RangeError: Maximum call stack size exceeded.`

Below is some sample code that returns error when parent-el is instantiated. 

```
<dom-module id="child-el">
  <template>{{value}}</template>
</dom-module>
<script>
(function() {
  Polymer({
    is: 'child-el',
    properties: {
      value: {
        value: null,
        notify: true
      }
    }
  });
}());
</script>

<dom-module id="parent-el">
  <template><child-el value="{{numvalue}}"></child-el></template>
</dom-module>
<script>
(function() {
  Polymer({
    is: 'parent-el',
    properties: {
      numvalue: {
        value: NaN,
        notify: true
      }
    }
  });
}());
</script>
```

### PR DESCRIPTION
**Pull request by azakus**
Fixes #1799

### CODE REVIEW

**Comment by arodic** [comment]
Fixes #1799?

File: src/lib/bind/accessors.html [hunk 1]
```diff
@@ -48,7 +48,10 @@
 
       _propertySet: function(property, value, effects) {
         var old = this.__data__[property];
-        if (old !== value) {
+        // NaN is always not equal to itself,
+        // if old and value are both NaN we treat them as equal
+        // x !== x is 10x faster, and equivalent to isNaN(x)
+        if (old !== value && !(old !== old && value !== value)) {
```
Comment (kevinpschaaf) [inline src/lib/bind/accessors.html]: Seems like `(old !== value && old==old && value==value)` is slightly more readable.

**Comment by azakus** [comment]
Tests are passing in VMs, not sure what is going on in Sauce.

**Comment by kevinpschaaf** [comment]
LGTM, confirmed min perf impact

**Pull request merged**